### PR TITLE
[cert-manager] Update cert-manager to v1.19.3

### PR DESCRIPTION
**Pull request by myasnikovdaniil**
### Release note

```release-note
[cert-manager] Updated cert-manager to v1.19.3
```

## Summary by CodeRabbit

* **New Features**
  * Global nodeSelector and hostUsers (pod user-namespace isolation) added
  * New/updated CRDs for cert-manager resources (Certificate, CertificateRequest, Order, etc.)

* **Documentation**
  * Revised chart docs and installation guidance; added deprecation/notice about private-key rotation
  * Removed legacy CRD README and schema files from the CRD package (documentation consolidated)

* **Chores**
  * Upgraded cert-manager to v1.19.3
  * Moved CRDs into a dedicated CRD package; ServiceMonitor targetPort default renamed to "http-metrics"

### CODE REVIEW

File: packages/system/cert-manager-crds/Makefile
Comment (kvaps) [inline packages/system/cert-manager-crds/Makefile]: Let's remove this section from `cert-manager-crds` it should be in sync with `cert-manager`

Comment (kvaps) [reply packages/system/cert-manager-crds/Makefile]: Please also update `packages/system/cert-manager/Makefile` to add something like this:

```
mv charts/cert-manager/templates/crds.yaml ../piraeus-operator-crds/templates/crds.yaml
```

Follow the pattern of piraeus-operator

https://github.com/cozystack/cozystack/blob/7ff5b2ba236ab7b82fd2ca794b4741352b8cd731/packages/system/piraeus-operator/Makefile#L11

Comment (myasnikovdaniil) [reply packages/system/cert-manager-crds/Makefile]: I agree, your approach will be more correct

**Review by lexfrei — CHANGES_REQUESTED** [review]
## E2E Test Failure: `kubernetes-latest` (v1.33) — Nodes Never Become Ready

### Test Failure Summary

The `kubernetes-latest` E2E test **consistently fails** (3 attempts, all fail). The `kubernetes-previous` (v1.32) test passes on this same PR. Both tests pass on main.

| cert-manager | tenant K8s | Node Ready time | Result |
|---|---|---|---|
| v1.17.2 (main) | v1.33 | ~71s | PASS |
| v1.17.2 (main) | v1.32 | ~95s | PASS |
| v1.19.3 (this PR) | v1.33 | >120s, never | **FAIL** |
| v1.19.3 (this PR) | v1.32 | ~83s | PASS |

The management cluster is K8s v1.33.2 in both cases. cert-manager is **disabled** in the tenant cluster (`certManager: enabled: false`).

**What happens:** The tenant v1.33 control plane comes up fully Ready (KamajiControlPlane INITIALIZED=true, READY=true, all deployments available, MachineDeployment replicas=2). Worker nodes **appear** (2 nodes found), but they **never** become Ready — even after 7+ minutes across 3 retries. This is not a flaky test or timeout edge case — something is fundamentally broken.

Refs:
- [Failed CI run](https://github.com/cozystack/cozystack/actions/runs/22154554864/job/64128228756)
- [Passing run on main (22149139788)](https://github.com/cozystack/cozystack/actions/runs/22149139788)

---

### cert-manager v1.17 → v1.19 Breaking Changes

This update skips v1.18, which introduced significant behavioral changes:

**v1.17 → v1.18:**
- `PrivateKeyRotationPolicy` default changed from `Never` to `Always` (feature gate `DefaultPrivateKeyRotationPolicyAlways`). Any certificate renewal now generates a **new private key**. For CA certificates, this can break the entire trust chain if downstream certs aren't re-issued.
- `RevisionHistoryLimit` default: nil → 1. Old CertificateRequest resources are auto-deleted.
- RSA signing: SHA-256 → SHA-384 for ≥3072-bit keys, SHA-512 for ≥4096-bit keys.

**v1.18 → v1.19:**
- CRD-based API defaults for `issuerRef`: `group` now defaults to `cert-manager.io`, `kind` to `Issuer`. This caused mass certificate re-issuance ([#8158](https://github.com/cert-manager/cert-manager/issues/8158), [#8182](https://github.com/cert-manager/cert-manager/issues/8182)).
- v1.19.3 fixes infinite re-issuance loop when issuer returns cert with mismatched public key.
- SKI (Subject Key Identifier) generation changed due to Go runtime update ([#8289](https://github.com/cert-manager/cert-manager/issues/8289)).

---

### Affected Certificate Resources in Cozystack

All webhook Certificate resources in the management cluster define `issuerRef` **without** the `group` field:

**Kamaji** (`packages/system/kamaji/charts/kamaji/templates/certmanager_certificate.yaml`):
```yaml
issuerRef:
  kind: Issuer
  name: kamaji-selfsigned-issuer
  # group: NOT specified
```

**CAPI Core** (`packages/system/capi-providers-core/files/core-components.yaml`):
```yaml
issuerRef:
  kind: Issuer
  name: capi-selfsigned-issuer
  # group: NOT specified
```

CAPI Bootstrap and Infrastructure providers follow the same pattern.

With v1.19 CRDs, the API server adds `group: cert-manager.io` automatically. Combined with `PrivateKeyRotationPolicy: Always`, this could trigger webhook certificate re-issuance with new keys, temporarily or permanently breaking webhook availability for Kamaji and CAPI providers.

---

### Values Configuration

```yaml
# Old (main) - correctly nested under subchart name:
cert-manager:
  installCRDs: false

# New (this PR) - NOT nested under subchart:
crds:
  enabled: true
```

The `crds: enabled: true` is at the top level of the umbrella chart, not nested under `cert-manager:`. The subchart receives only its default values. Functionally this works since CRDs are in a separate package and the only override was `installCRDs: false`, but the structure is misleading.

Also: `_helpers.tpl` in the CRDs package references `.Values.creator` which is not defined in `cert-manager-crds/values.yaml` (it comes from the cert-manager chart defaults where it equals `"helm"`).

---

### Recommended Next Steps

1. **Debug the v1.33 node readiness failure** — check in the E2E environment:
   - `kubectl describe node` — which conditions are not met
   - KCCM logs — is it initializing v1.33 nodes
   - Cilium HelmRelease status — is CNI being deployed to the tenant cluster
   - cert-manager Certificate status — any unexpected re-issuance
   - Kamaji/CAPI webhook health

2. **Consider pinning `privateKey.rotationPolicy: Never`** for webhook certificates (Kamaji, CAPI providers)

3. **Add explicit `group: cert-manager.io`** to `issuerRef` in Kamaji/CAPI Certificate resources to avoid CRD defaulting side effects

4. **Ensure kubernetes-latest E2E passes** before merging

File: packages/system/cert-manager/charts/cert-manager/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ annotations:
     fingerprint: 1020CF3C033D4F35BAE1C19E1226061C665DF13E
     url: https://cert-manager.io/public-keys/cert-manager-keyring-2021-09-20-1020CF3C033D4F35BAE1C19E1226061C665DF13E.gpg
 apiVersion: v2
-appVersion: v1.17.2
+appVersion: v1.19.3
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply packages/system/cert-manager/charts/cert-manager/Chart.yaml:9]: CRDs installed separately in cert-manager-crds chart

File: packages/system/cert-manager/charts/cert-manager/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -12,7 +12,8 @@ metadata:
   {{- with .Values.serviceAccount.annotations }}
   annotations:
     {{- range $k, $v := . }}
-      {{- printf "%s: %s" (tpl $k $) (tpl $v $) | nindent 4 }}
+      {{- $value := $v | quote }}
+      {{- printf "%s: %s" (tpl $k $) (tpl $value $) | nindent 4 }}
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply packages/system/cert-manager/charts/cert-manager/templates/serviceaccount.yaml:16]: We don't fix upstream chart code, so this issue is irrelevant

File: packages/system/cert-manager/charts/cert-manager/templates/deployment.yaml [hunk 3]
```diff
@@ -66,6 +66,9 @@ spec:
       {{- with .Values.global.priorityClassName }}
       priorityClassName: {{ . | quote }}
       {{- end }}
+      {{- if (hasKey .Values.global "hostUsers") }}
+      hostUsers: {{ .Values.global.hostUsers }}       
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply packages/system/cert-manager/charts/cert-manager/templates/deployment.yaml:71]: We don't fix upstream chart code, so this issue is irrelevant

File: hack/e2e-apps/run-kubernetes.sh [hunk 4]
```diff
@@ -98,15 +98,18 @@ EOF
     done
   '
   # Verify the nodes are ready
-  kubectl --kubeconfig "tenantkubeconfig-${test_name}" wait node --all --timeout=2m --for=condition=Ready
+  kubectl --kubeconfig "tenantkubeconfig-${test_name}" wait node --all --timeout=2m --for=condition=Ready || true
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply hack/e2e-apps/run-kubernetes.sh]: Fixed

File: hack/e2e-apps/run-kubernetes.sh [hunk 5]
```diff
@@ -98,15 +98,18 @@ EOF
...
   kubectl --kubeconfig "tenantkubeconfig-${test_name}" get nodes -o wide
+  kubectl --kubeconfig "tenantkubeconfig-${test_name}" describe nodes
+  kubectl -n tenant-test get hr  # Cilium/CoreDNS HelmRelease status
+  kubectl -n tenant-test get certificates  # cert-manager certificate health
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply hack/e2e-apps/run-kubernetes.sh]: This won't fail because commands check parent cluster, not nested

File: hack/e2e-apps/run-kubernetes.sh
Comment (lllamnyp) [inline hack/e2e-apps/run-kubernetes.sh:1]: I don't think this changed is related to the PRs intent.

File: packages/system/cert-manager-crds/Makefile
Comment (lllamnyp) [inline packages/system/cert-manager-crds/Makefile:1]: Previously `make update` would download the latest CRDs directly from upstream as a subchart. Now the CRDs are simply statically placed into the wrapper chart.

File: packages/system/cert-manager/Makefile
Comment (lllamnyp) [inline packages/system/cert-manager/Makefile:1]: This is confusing. This Makefile is now responsible for updating both the cert-manager chart **and** the cert-manager-crds chart.

File: packages/system/cert-manager/values.yaml
Comment (lllamnyp) [inline packages/system/cert-manager/values.yaml:1]: This is incorrect and will fail installation. This key should be nested under the `cert-manager` key and `crd.enabled` should be false (as it's installed by the `cert-manager-crds` chart, not here).

Comment (myasnikovdaniil) [reply packages/system/cert-manager/values.yaml:1]: This won't fail the installation, because crds templates are removed from this chart to crds chart. But values are incorrect, it's my fault.

File: packages/system/cert-manager/Makefile
Comment (myasnikovdaniil) [reply packages/system/cert-manager/Makefile:1]: It was done intentionally to have single point of updating app, like it already done in piraeus operator. This way developer less likely get different versions of crds and operator code.

File: packages/system/cert-manager/Makefile [hunk 6]
```diff
@@ -8,3 +8,5 @@ update:
 	helm repo add jetstack https://charts.jetstack.io
 	helm repo update jetstack
 	helm pull jetstack/cert-manager --untar --untardir charts
+	mv charts/cert-manager/templates/crd-*.yaml ../cert-manager-crds/templates/
```
Comment (lexfrei) [inline packages/system/cert-manager/Makefile:12]: `mv` and `cp` add files to `../cert-manager-crds/templates/` but never clean up existing ones first. If a future upstream release renames or removes a CRD file, stale files will remain and continue to be rendered.

Add a cleanup step before `mv`:
```makefile
	rm -f ../cert-manager-crds/templates/crd-*.yaml ../cert-manager-crds/templates/_helpers.tpl
	mv charts/cert-manager/templates/crd-*.yaml ../cert-manager-crds/templates/
	cp charts/cert-manager/templates/_helpers.tpl ../cert-manager-crds/templates/
```

File: packages/system/cert-manager/charts/cert-manager/Chart.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply packages/system/cert-manager/charts/cert-manager/Chart.yaml:9]: We did not edit upstream chart directly, it was done via makefile update script.

File: packages/system/cert-manager/charts/cert-manager/values.schema.json [hunk 7]
```diff
@@ -718,6 +724,10 @@
...
       "type": "object"
     },
+    "helm-values.global.hostUsers": {
+      "description": "Set all pods to run in a user namespace without host access. Experimental: may be removed once the Kubernetes User Namespaces feature is GA.\n\nRequirements:\n  - Kubernetes ≥ 1.33, or\n  - Kubernetes 1.27–1.32 with UserNamespacesSupport feature gate enabled.\n\nSet to false to run pods in a user namespace without host access.\n\nSee [limitations](https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#limitations) for details.",
+      "type": "boolean"
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply packages/system/cert-manager/charts/cert-manager/values.schema.json:729]: We don't fix issues in upstream charts

File: packages/system/cert-manager/charts/cert-manager/values.yaml [hunk 8]
```diff
@@ -28,6 +38,19 @@ global:
...
+  # Set all pods to run in a user namespace without host access.
+  # Experimental: may be removed once the Kubernetes User Namespaces feature is GA.
+  #
+  # Requirements:
+  #   - Kubernetes ≥ 1.33, or
+  #   - Kubernetes 1.27–1.32 with UserNamespacesSupport feature gate enabled.
+  #
+  # Set to false to run pods in a user namespace without host access.
+  #
+  # See [limitations](https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#limitations) for details.
```
[the remark this answers is not in the document]
Comment (myasnikovdaniil) [reply packages/system/cert-manager/charts/cert-manager/values.yaml:50]: We don't fix issues in upstream helm charts.

File: packages/system/cert-manager/Makefile (hunk 6 above)
Comment (myasnikovdaniil) [reply packages/system/cert-manager/Makefile:12]: added less specific cleanup for crds, anyway if upstream cert-manager chart will change its structure again (filenames, etc.) than we'll need to refactor this script again

**Review by lexfrei — APPROVED** [review]
